sqlalchemy-stubs/engine/default.pyi

```diff
@@ -68,119 +86,148 @@ class DefaultDialect(interfaces.Dialect):
+        self, dbapi_connection: interfaces._DBAPIConnection
+    ) -> None: ...
+    def do_rollback(
+        self, dbapi_connection: interfaces._DBAPIConnection
+    ) -> None: ...
+    def do_commit(
+        self, dbapi_connection: interfaces._DBAPIConnection
+    ) -> None: ...
+    def do_close(
+        self, dbapi_connection: interfaces._DBAPIConnection
+    ) -> None: ...
+    def do_ping(
+        self, dbapi_connection: interfaces._DBAPIConnection
+    ) -> bool: ...
+    def create_xid(self) -> str: ...
+    def do_savepoint(self, connection: base.Connection, name: str) -> None: ...
+    def do_rollback_to_savepoint(
+        self, connection: base.Connection, name: str
+    ) -> None: ...
+    def do_release_savepoint(
+        self, connection: base.Connection, name: str
+    ) -> None: ...
     def do_executemany(
         self,
-        cursor: Any,
+        cursor: interfaces._DBAPICursor,
         statement: Any,
         parameters: Any,
-        context: Optional[Any] = ...,
+        context: Optional[interfaces.ExecutionContext] = ...,
     ) -> None: ...
     def do_execute(
         self,
-        cursor: Any,
+        cursor: interfaces._DBAPICursor,
         statement: Any,
         parameters: Any,
-        context: Optional[Any] = ...,
+        context: Optional[interfaces.ExecutionContext] = ...,
     ) -> None: ...
-    def do_execute_no_params(
-        self, cursor: Any, statement: Any, context: Optional[Any] = ...
+    def do_execute_no_params(  # type: ignore[override]
+        self,
+        cursor: interfaces._DBAPICursor,
+        statement: Any,
+        context: Optional[interfaces.ExecutionContext] = ...,
+    ) -> None: ...
+    def is_disconnect(
+        self,
+        e: Any,
+        connection: base.Connection,
+        cursor: interfaces._DBAPICursor,
+    ) -> bool: ...
+    def reset_isolation_level(
+        self, dbapi_conn: interfaces._DBAPIConnection
     ) -> None: ...
-    def is_disconnect(self, e: Any, connection: Any, cursor: Any): ...
-    def reset_isolation_level(self, dbapi_conn: Any) -> None: ...
-    def normalize_name(self, name: Any): ...
-    def denormalize_name(self, name: Any): ...
+    def normalize_name(self, name: Optional[str]) -> Optional[str]: ...
+    def denormalize_name(self, name: Optional[str]) -> Optional[str]: ...
 
-class _RendersLiteral:
-    def literal_processor(self, dialect: Any): ...
+class _RendersLiteral(Generic[_T]):
+    def literal_processor(
+        self, dialect: interfaces.Dialect
+    ) -> _LiteralProcessor[_T]: ...
 
-class _StrDateTime(_RendersLiteral, sqltypes.DateTime): ...
-class _StrDate(_RendersLiteral, sqltypes.Date): ...
-class _StrTime(_RendersLiteral, sqltypes.Time): ...
+class _StrDateTime(_RendersLiteral[datetime], sqltypes.DateTime): ...  # type: ignore[misc]
+class _StrDate(_RendersLiteral[date], sqltypes.Date): ...  # type: ignore[misc]
+class _StrTime(_RendersLiteral[time], sqltypes.Time): ...  # type: ignore[misc]
```

Comment:
I think these _StrDateTime, _StrDate, _StrTime can be removed, since they are internal